beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,166 @@
+# is not comprehensive (missing data, grad, grad_fn, shape). Contains
+# ``memory_format`` which is not an attribute
 #
-# You can use standard NumPy-like indexing with all bells and whistles!
 
-print(x[:, 1])
+tnsr = torch.rand(3,4)
 
-###############################################################
-# Resizing: If you want to resize/reshape tensor, you can use ``torch.view``:
-x = torch.randn(4, 4)
-y = x.view(16)
-z = x.view(-1, 8)  # the size -1 is inferred from other dimensions
-print(x.size(), y.size(), z.size())
+print(f"Data stored in tensor:\n{tnsr.data}\n")
+print(f"Shape of tensor: {tnsr.shape}")
+print(f"Datatype of tensor: {tnsr.dtype}")
+print(f"Device tensor lives on: {tnsr.device}")
 
-###############################################################
-# If you have a one element tensor, use ``.item()`` to get the value as a
-# Python number
-x = torch.randn(1)
-print(x)
-print(x.item())
+print("-------")
 
-###############################################################
-# **Read later:**
-#
-#
-#   100+ Tensor operations, including transposing, indexing, slicing,
-#   mathematical operations, linear algebra, random numbers, etc.,
-#   are described
-#   `here <https://pytorch.org/docs/torch>`_.
+# Autograd-related attributes; more on this later
+print(f"Is tensor a leaf on computational graph: {tnsr.is_leaf}")
+print(f"Does tensor require gradients: {tnsr.requires_grad}")
+print(f"Accumulated gradients of tensor: {tnsr.grad}")
+print(f"Function that computed this tensor's gradient: {tnsr.grad_fn}")
+
+
+
+######################################################################
+# --------------
 #
-# NumPy Bridge
-# ------------
+
+
+######################################################################
+# Tensor Operations
+# ~~~~~~~~~~~~~~~~~
 #
-# Converting a Torch Tensor to a NumPy array and vice versa is a breeze.
+# Over 100 tensor operations, including transposing, indexing, slicing,
+# mathematical operations, linear algebra, random sampling, and more are
+# comprehensively described
+# `here <https://pytorch.org/docs/stable/torch.html>`__.
 #
-# The Torch Tensor and NumPy array will share their underlying memory
-# locations (if the Torch Tensor is on CPU), and changing one will change
-# the other.
+# Each of them can be run on the GPU (at typically higher speeds than on a
+# CPU). If you’re using Colab, allocate a GPU by going to Edit > Notebook
+# Settings.
 #
-# Converting a Torch Tensor to a NumPy Array
-# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-a = torch.ones(5)
-print(a)
+# We move our tensor to the GPU if available
+if torch.cuda.is_available:
+  tnsr.to('cuda')
 
-###############################################################
-#
 
-b = a.numpy()
-print(b)
+######################################################################
+# Try out some of the operations from the list. You may come across multiple
+# syntaxes for the same operation. Don’t get confused by the aliases!
```

Comment:
I'm not sure what you refer to as aliases here?

Reply:
I think it's fine to mention that some ops can have different names but I would still keep all the discussion around numpy arrays as most users these days come from numpy and heavily use numpy around pytorch!